crates/dojo-signer/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use starknet::signers::{LocalWallet, VerifyingKey, SigningKey, Signer};
+use starknet::core::{crypto::{Signature}, types::FieldElement};
+use dotenv::dotenv;
+use lazy_static::lazy_static;
+use std::sync::Mutex;
+use std::env;
+use async_trait::async_trait;
+
+lazy_static! {
+    static ref STARK_PRIVATE_KEY: Mutex<Option<SigningKey>> = Mutex::new(None);
+}
+
+#[derive(Debug, Clone)]
+pub struct EnvLocalWallet {
+    inner_wallet: LocalWallet,
+}
+
+impl EnvLocalWallet {
+    pub fn from_env() -> Result<Self, Box<dyn std::error::Error>> {
+        dotenv().ok();
+
+        let private_key = {
+            let mut private_key_guard = STARK_PRIVATE_KEY.lock().unwrap();
```

Comment:
Hm how come a mutex is required here?

Reply:
was just from an example I was working off - I can remove